test/integration/deploy_trigger_test.go

```diff
@@ -192,14 +194,18 @@ waitForNewConfig:
 		case event := <-configWatch.ResultChan():
 			if event.Type == watchapi.Modified {
 				newConfig = event.Object.(*deployapi.DeploymentConfig)
-				break waitForNewConfig
+				// Multiple updates to the config can be expected (e.g. status
+				// updates), so wait for a significant update (e.g. version).
+				if newConfig.LatestVersion > 0 {
```

Comment:
if we never see it, the test will just timeout, right? do we want to set a timeout on that earlier than the integration test timeout?